trace.go

```diff
@@ -34,63 +36,69 @@ type traceExporter struct {
 	projectID string
 	bundler   *bundler.Bundler
 	// uploadFn defaults to uploadSpans; it can be replaced for tests.
-	uploadFn func(spans []*trace.SpanData)
+	uploadFn func(spans []*tracepb.Span)
 	overflowLogger
 	client *tracingclient.Client
 }
 
 var _ trace.Exporter = (*traceExporter)(nil)
 
 func newTraceExporter(o Options) (*traceExporter, error) {
-	client, err := tracingclient.NewClient(o.Context, o.TraceClientOptions...)
+	ctx := o.Context
+	if ctx == nil {
+		ctx = context.Background()
+	}
+	client, err := tracingclient.NewClient(ctx, o.TraceClientOptions...)
 	if err != nil {
 		return nil, fmt.Errorf("stackdriver: couldn't initialize trace client: %v", err)
 	}
 	return newTraceExporterWithClient(o, client), nil
 }
 
+const defaultBufferedByteLimit = 8 * 1024 * 1024
+
 func newTraceExporterWithClient(o Options, c *tracingclient.Client) *traceExporter {
 	e := &traceExporter{
 		projectID: o.ProjectID,
 		client:    c,
 		o:         o,
 	}
-	bundler := bundler.NewBundler((*trace.SpanData)(nil), func(bundle interface{}) {
-		e.uploadFn(bundle.([]*trace.SpanData))
+	b := bundler.NewBundler((*tracepb.Span)(nil), func(bundle interface{}) {
+		e.uploadFn(bundle.([]*tracepb.Span))
 	})
 	if o.BundleDelayThreshold > 0 {
-		bundler.DelayThreshold = o.BundleDelayThreshold
+		b.DelayThreshold = o.BundleDelayThreshold
 	} else {
-		bundler.DelayThreshold = 2 * time.Second
+		b.DelayThreshold = 2 * time.Second
```

Comment:
Now that we have a defined default threshold/time, it'd be nice to remove this magic value

Reply:
Well this isn't really related to the timeout